### PR TITLE
NN-1886: Add in enable / disable auth user

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.java [hunk 1]
```diff
@@ -62,4 +69,23 @@ public StaffUserAccount getUserByExternalIdentifier(final String idType, final S
...
+        changeUserStatus(username, false, admin);
+    }
+
+    private void changeUserStatus(final String username, final boolean status, final String admin) {
+        final var userEmail = getAuthUserByUsername(username)
+                .orElseThrow(() -> new EntityNotFoundException(String.format("User not found with username %s", username)));
+        userEmail.setEnabled(status);
+        userEmailRepository.save(userEmail);
+        telemetryClient.trackEvent("AuthUserChangeStatus",
+                Map.of("username", username, "status", Boolean.toString(status), "admin", admin), null);
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.java]: `enabled` might be a bit more descriptive than `status`?

File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserServiceTest.java [hunk 2]
```diff
@@ -70,6 +76,54 @@ public void findByEmailAndMasterIsTrue() {
...
+        verify(telemetryClient).trackEvent("AuthUserChangeStatus", Map.of("username", "someuser", "admin", "someadmin", "status", "true"), null);
+    }
+
+    @Test
+    public void disableUser() {
+        final var optionalUserEmail = createUserEmailUser();
+        when(userEmailRepository.findByUsernameAndMasterIsTrue(anyString())).thenReturn(optionalUserEmail);
+        userService.disableUser("user", "admin");
+        assertThat(optionalUserEmail).get().extracting(UserEmail::isEnabled).isEqualTo(Boolean.FALSE);
+        //noinspection OptionalGetWithoutIsPresent
```
Comment (andrewrlee) [inline src/test/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserServiceTest.java]: meh.. could you remove the warning with [orElseThrow()](https://docs.oracle.com/javase/10/docs/api/java/util/Optional.html#orElseThrow())? or does the warning apply to that too?

Comment (petergphillips) [reply src/test/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserServiceTest.java]: Good idea, done

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.java (hunk 1 above)
Comment (petergphillips) [reply src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.java]: Agreed, done

**Pull request merged**